### PR TITLE
Some grabbing tweaks

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Aggressive grabs from pacifists now make people drop their items, just like aggressive grabs from non-pacifists do.

I also altered the formula for calculating your chance to resist out of a grab to `max((resist_chance/altered_grab_state)-sqrt((getBruteLoss()+getFireLoss()+getOxyLoss()+getToxLoss()+getCloneLoss())*0.5+getStaminaLoss()), 0)`.

This should make how damaged you are actually affect your chances for breaking out of a neck grab, which in turn should mean that a neck grab on someone who's heavily damaged won't sometimes be HARDER to break out of than a kill grab on that person (as far as I can tell, the last column of https://i.imgur.com/6yAT90T.png is wrong, as the base chance per attempt to break out of a kill grab under the current system should be 10%, not 5%).

I also added some parenthesis and moved some stuff around to make the order of operations/logic of the equation more clear (hopefully)

## Why It's Good For The Game

Making someone drop their held items doesn't damage them, and pacifists are allowed to use things like stun batons, so I don't see why they shouldn't be allowed to make people drop their items when they aggressively grab them (which is what aggressive grabs from non-pacifists already do).

The change to the chance to break out of a grab will fix an unintuitive oddity and hopefully make choking someone out more viable when you don't have cuffs or the like. Someone needs to basically be helpless for you to put them in a kill grab anyway, so I felt like it shouldn't be easy to break out of it if you're not in cuffs or stunned.

## Changelog
:cl: ATHATH
balance: Aggressive grabs from pacifists now make people drop their items, just like aggressive grabs from non-pacifists do.
balance: The chances to break out of aggressive, neck, and kill grabs have been modified slightly, and having large amounts of damage should actually affect how hard it is to break out of a kill grab now.
balance: All types of damage now affect how hard it is to break out of a grab, not just brute and stamina damage. Stamina damage is, as before, weighted twice as much as the other damage types in the grab escape chance calculation(s).
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -759,7 +759,7 @@
 		if((resting || HAS_TRAIT(src, TRAIT_GRABWEAKNESS)) && pulledby.grab_state < GRAB_KILL) //If resting, resisting out of a grab is equivalent to 1 grab state higher. wont make the grab state exceed the normal max, however
 			altered_grab_state++
 		var/resist_chance = BASE_GRAB_RESIST_CHANCE // see defines/combat.dm
-		resist_chance = max(resist_chance/altered_grab_state-sqrt((getStaminaLoss()+getBruteLoss()/2)*(3-altered_grab_state)), 0) // https://i.imgur.com/6yAT90T.png for sample output values
+		resist_chance = max((resist_chance/altered_grab_state)-sqrt(((getBruteLoss()/2)+getStaminaLoss())*altered_grab_state), 0)
```
Comment (Qustinnus) [inline code/modules/mob/living/living.dm]: Update the picture in the comment please.

**Comment by ATH1909** [comment]
@Qustinnus Here's the new chart you asked for:
chart1

**Comment by ATH1909** [comment]
And here's a chart where the formula's changed to be just max((resist_chance/altered_grab_state)-sqrt((getBruteLoss()/2)+getStaminaLoss()), 0):
chart2

**Comment by ATH1909** [comment]
And finally, here's a chart of what the grab escape chances ACTUALLY are under our current (before this PR) system/equation for them:
chart3

File: code/modules/mob/living/living.dm (hunk 1 above)
Comment (spookydonut) [inline code/modules/mob/living/living.dm]:
```suggestion
		resist_chance = max((resist_chance/altered_grab_state)-sqrt(((getBruteLoss()*0.5)+getStaminaLoss())*altered_grab_state), 0)
```

**Comment by PKPenguin321** [comment]
Weird math error on my part, sorry about that

**Comment by PKPenguin321** [comment]
Also as for making the calc use all damage, that would probably be fine. I only used brute because I wanted to keep it in scope with my punching rebalance at the time but you can change it if you want

**Comment by ExcessiveUseOfCobblestone** [comment]
I would prefer there be no 0%, there should at least be SOME chance to break out even if it's not likely.

**Comment by ATH1909** [comment]
> I would prefer there not be a 0%, there should at least be SOME chance to break out even if it's not likely.
> 
> If you wanna restrain someone "permanently" we have things like cuffs for that.

Eh, I'd prefer it if we had options that could reasonably restrain someone other than cuffs, because not everyone rushes to make cablecuffs/grab cable coils at roundstart. Regardless, a "kill" grab isn't a "permanent" restraint, it's a kill method- more like beating someone into crit than cuffing them.

**Comment by ExcessiveUseOfCobblestone** [comment]
having like a 1 prob is pretty reasonable considering this involves no gear whatsoever.

**Comment by ATH1909** [comment]
Alright, I've changed the formula to be max((resist_chance/altered_grab_state)-sqrt((getBruteLoss()+getFireLoss()+getOxyLoss()+getToxLoss()+getCloneLoss())*0.5+getStaminaLoss()), 0).

Basically, burn, oxyloss, tox, and clone damage are now factored into (and weighted the same amount as brute damage in) the grab escape chance calculation(s). Stamina damage still counts, of course, and is weighted twice as heavily as brute damage is in those calculation(s).

Also, the chance to escape a graph now adheres to this chart (the second one I posted, not the first one), provided that you replace "brute damage" with "the sum of all of the victim's non-stamina damage": https://user-images.githubusercontent.com/42606352/75658908-b6012200-5c2e-11ea-9dd9-e4fbd475c7a5.PNG

To get down to a 0% chance to escape, they'd have to have 100 stamina damage, 200 non-stamina damage, or some mixture of the two (say, 75 stamina damage and 50 brute damage).

@ExcessiveUseOfCobblestone Is this new formula satisfactory to you?

**Comment by ExcessiveUseOfCobblestone** [comment]
god what a complex and annoying formula.

**Comment by ATH1909** [comment]
Thanks.

**Pull request merged**